app/controllers/indicators_controller.rb

```diff
@@ -72,6 +73,18 @@ def set_indicator_data
     @summary_value = calculations[:summary_value]
   end
 
+  def set_coin_stats
+    @coin_stats = Rails.cache.fetch("indicators/#{@coin.slug}:stats") do
+      fetched_stats = CoinServices::UpdateIndicatorStats.call.result
+
+      if fetched_stats[@coin.slug].present?
```

Comment:
I *think* you can further simplify this entire `if/else` block to just

```
    fetched_stats.fetch(@coin.slug) || nil
```

Might be good to protect against the case where the coin slug doesn't exist in `fetched_stats` for whatever reason.

Reply:
But that said maybe we do want to throw an exception for Rollbar to pick up to alert us about the problem...